include/pros/imu.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * \file pros/imu.h
+ *
+ * Contains prototypes for functions related to the VEX Inertial sensor.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/imu.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2019, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_IMU_H_
+#define _PROS_IMU_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+namespace c {
+#endif
+
+typedef enum imu_status_e {
+	E_IMU_STATUS_CALIBRATING = 0x01,
+	E_IMU_STATUS_ERROR = 0xFF,  // NOTE: used for returning an error from the get_status function, not that the IMU is
+	                            // necessarily in an error state
+} imu_status_e_t;
+
+typedef struct __attribute__((__packed__)) quaternion_s {
+	double a;
+	double b;
+	double c;
+	double d;
+} quaternion_s_t;

```

Comment:
Quaternions are comprised of a vector component and a scalar component, usually denoted by `x, y, z` and `w`. Should this be renamed, or does vexOS do something weird with the data?

Reply:
wikipedia told me
> _a_ + _b_**i** + _c_**j** + _d_**k**

so i just went with that, but i'm willing to change it (back) to w/x/y/z if that's more correct

Reply:
Is there a difference between this and imu_s_t besides the layout? We could combine them on the pros API side.

Reply:
the difference is semantic

Reply:
.... which I suppose could just be handled with a typedef

Reply:
I'm more familiar with w,x,y,z for quaternions, and I would advocate for that naming only because the (x,y,z) component is a bit more obvious in that form than the a,b,c,d format. I don't think it's a big deal which version we go with, but that's my $0.02